### PR TITLE
Refactor constructor delegation

### PR DESCRIPTION
**Pull request by knutwannheden**
Add new `K.ConstructorInvocation` type and `K.Constructor` to be used for constructor delegation when specified in class header or on any other constructor.

### CODE REVIEW

**Comment by knutwannheden** [comment]
@traceyyoshima I think this can be reviewed now.

File: src/main/java/org/openrewrite/kotlin/KotlinIsoVisitor.java [hunk 1]
```diff
@@ -32,6 +32,16 @@ public K.Binary visitBinary(K.Binary binary, P p) {
         return (K.Binary) super.visitBinary(binary, p);
     }
 
+    @Override
+    public K.Constructor visitConstructor(K.Constructor constructor, P p) {
+        return (K.Constructor) super.visitConstructor(constructor, p);
+    }
+
+    @Override
+    public K.ConstructorInvocation visitConstructorInvocation(K.ConstructorInvocation destructuringDeclaration, P p) {
```
Comment (traceyyoshima) [inline src/main/java/org/openrewrite/kotlin/KotlinIsoVisitor.java:41]:
```suggestion
    public K.ConstructorInvocation visitConstructorInvocation(K.ConstructorInvocation constructorInvocation, P p) {
```

File: src/main/java/org/openrewrite/kotlin/KotlinIsoVisitor.java [hunk 2]
```diff
@@ -32,6 +32,16 @@ public K.Binary visitBinary(K.Binary binary, P p) {
...
+        return (K.ConstructorInvocation) super.visitConstructorInvocation(destructuringDeclaration, p);
```
Comment (traceyyoshima) [inline src/main/java/org/openrewrite/kotlin/KotlinIsoVisitor.java:42]:
```suggestion
        return (K.ConstructorInvocation) super.visitConstructorInvocation(constructorInvocation, p);
```

**Review by traceyyoshima — APPROVED** [review]
Looks good! I'm curious about what your thoughts are on whether we should be adding SignatureBuilding and TypeMapping tests as the parsing is improved?

**Review by kunli2 — APPROVED** [review]

**Pull request merged**